enderio-machines/src/main/java/com/enderio/machines/common/blockentity/EnchanterBlockEntity.java

```diff
@@ -0,0 +1,60 @@
+package com.enderio.machines.common.blockentity;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.enderio.machines.common.blockentity.data.sidecontrol.item.ItemHandlerMaster;
+import com.enderio.machines.common.menu.EnchanterMenu;
+import com.enderio.machines.common.recipe.IEnchanterRecipe;
+import com.enderio.machines.common.recipe.MachineRecipes;
+
+import lombok.Getter;
+import net.minecraft.core.BlockPos;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraftforge.items.wrapper.RecipeWrapper;
+
+public class EnchanterBlockEntity extends AbstractMachineBlockEntity{
+
+    @Getter
+    private ItemHandlerMaster itemHandlerMaster = new ItemHandlerMaster(getConfig(), 4, List.of(0,1,2), List.of(3)) {
+        protected void onContentsChanged(int slot) {
+            if (slot != 3) {
+                Optional<IEnchanterRecipe> recipe = level.getRecipeManager().getRecipeFor(MachineRecipes.Types.ENCHANTING, new RecipeWrapper(itemHandlerMaster), level);
+                if (recipe.isPresent()) {
+                    itemHandlerMaster.setStackInSlot(3, recipe.get().assemble(new RecipeWrapper(itemHandlerMaster)));
+                }
+                else {
+                    itemHandlerMaster.setStackInSlot(3, ItemStack.EMPTY);
+                }
+            }
+            setChanged();
```

Comment:
Changing the RedstoneControl doesn't call setChanged(). The setter for redstoneControl doesn't mark the BlockEntity as changed. The fix would be either to implement a setChanged() in the setter of setRedstoneControl or adding a tick function. You could also call it in the SyncClientToServerMenuPacket as it would be the only  setter for it with 
        getMenu(context).getBlockEntity() after doing a nonnullcheck on getBlockEntity as it's nullable
        
getMenu(context).getBlockEntity().setChanged();

Reply:
I've implemented redstone control in the itemhandler and slot, and added setchanged() to the setter.